src/solidity-frontend/solidity_convert.cpp

```diff
@@ -0,0 +1,1898 @@
+  typet type;
+  // Creat a new code_typet, parse the return_type and copy the code_typet to typet
+  code_typet convert_type;
+  typet return_type;
+  if(
+    decl["name"] == "assert" || decl["name"] == "__ESBMC_assume" ||
+    decl["name"] == "__VERIFIER_assume")
+  {
+    assert(decl["typeDescriptions"]["typeString"] == "function (bool) pure");
+    // clang's assert(.) uses "signed_int" as assert(.) type (NOT the argument type),
+    // while Solidity's assert uses "bool" as assert(.) type (NOT the argument type).
+    return_type = bool_type();
+    std::string c_type = "bool";
+    return_type.set("#cpp_type", c_type);
+    convert_type.return_type() = return_type;
+
+    if(!convert_type.arguments().size())
+      convert_type.make_ellipsis();
+  }
+  else
+  {
+    assert(!"Unsupported special functions");
+  }
+
+  type = convert_type;
+
+  new_expr = exprt("symbol", type);
+  new_expr.identifier(id);
+  new_expr.cmt_lvalue(true);
+  new_expr.name(name);
+  return false;
+}
+
+bool solidity_convertert::get_type_description(
+  const nlohmann::json &type_name,
+  typet &new_type)
+{
+  // For Solidity rule type-name:
+  SolidityGrammar::TypeNameT type = SolidityGrammar::get_type_name_t(type_name);
+
+  switch(type)
+  {
+  case SolidityGrammar::TypeNameT::ElementaryTypeName:
+  {
+    // rule state-variable-declaration
+    return get_elementary_type_name(type_name, new_type);
+  }
+  case SolidityGrammar::TypeNameT::ParameterList:
+  {
+    // rule parameter-list
+    // Used for Solidity function parameter or return list
+    return get_parameter_list(type_name, new_type);
+  }
+  case SolidityGrammar::TypeNameT::Pointer:
+  {
+    // auxiliary type: pointer (FuncToPtr decay)
+    // This part is for FunctionToPointer decay only
+    assert(
+      type_name["typeString"].get<std::string>().find("function") !=
+      std::string::npos);
+
+    // Since Solidity does not have this, first make a pointee
+    nlohmann::json pointee = make_pointee_type(type_name);
+    typet sub_type;
+    if(get_func_decl_ref_type(pointee, sub_type))
+      return true;
+
+    if(
+      sub_type.is_struct() ||
+      sub_type.is_union()) // for "assert(sum > 100)", false || false
+      assert(!"struct or union is NOT supported");
+
+    new_type = gen_pointer_type(sub_type);
+    break;
+  }
+  case SolidityGrammar::TypeNameT::PointerArrayToPtr:
+  {
+    // auxiliary type: pointer (FuncToPtr decay)
+    // This part is for FunctionToPointer decay only
+    assert(
+      type_name["typeIdentifier"].get<std::string>().find("ArrayToPtr") !=
+      std::string::npos);
+
+    // Array type descriptor is like:
+    //  "typeIdentifier": "ArrayToPtr",
+    //  "typeString": "uint8[2] memory"
+
+    // Since Solidity does not have this, first make a pointee
+    typet sub_type;
+    if(
+      get_array_to_pointer_type(
+        type_name,
+        sub_type)) // TODO: Fix me! Inconsistency. Better to use Solidity compiler library?
```

Comment:
Remember to open an issue

Reply:
ACK. 
will do it when extending the new frontend to support more benchmarks.